core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2TokenExchange.java

```diff
@@ -69,24 +72,33 @@ public void stop()
         executor.shutdownNow();
     }
 
-    public void setAccessToken(UUID authId, String accessToken)
+    @Override
+    public void setAccessToken(String authIdHash, String accessToken)
     {
-        cache.getUnchecked(authId).set(TokenPoll.token(accessToken));
+        cache.getUnchecked(authIdHash).set(TokenPoll.token(accessToken));
     }
 
-    public void setTokenExchangeError(UUID authId, String message)
+    @Override
+    public void setTokenExchangeError(String authIdHash, String message)
     {
-        cache.getUnchecked(authId).set(TokenPoll.error(message));
+        cache.getUnchecked(authIdHash).set(TokenPoll.error(message));
     }
 
     public ListenableFuture<TokenPoll> getTokenPoll(UUID authId)
     {
-        return nonCancellationPropagating(cache.getUnchecked(authId));
+        return nonCancellationPropagating(cache.getUnchecked(hashAuthId(authId)));
     }
 
     public void dropToken(UUID authId)
     {
-        cache.invalidate(authId);
+        cache.invalidate(hashAuthId(authId));
+    }
+
+    public static String hashAuthId(UUID authId)
```

Comment:
What kind of a security exploit is going to be mitigated by hashing a UUID v4?

Reply:
The real authId is never exposed to a browser so a compromised browser can not steal a token.